File: Elixir/led_numbers.exs
defmodule LED_Numbers do
  def showLED(num) when is_number(num) do
    digits = for x <- to_charlist(num) do
               x - 48
             end
    leds = %{0 => {" _  ","| | ","|_| "},
             1 => {"  ","| ","| "},
             2 => {" _  "," _| ","|_  "},
             3 => {"_  ","_| ","_| "},
             4 => {"    ","|_| ","  | "},
             5 => {" _  ","|_  "," _| "},
             6 => {" _  ","|_  ","|_| "},
             7 => {"_   "," |  "," |  "},
             8 => {" _  ","|_| ","|_| "},
             9 => {" _  ","|_| "," _| "}}
	build_output(digits,digits,leds,0,"")	     
  end
  
  def build_output(_,_,_,3,acc) do
    IO.puts(acc)
  end
  
  def build_output([],digits,leds,lineNo,acc) do
    build_output(digits,digits,leds,lineNo + 1,acc <> "\n")
  end
  
  def build_output([h|t],digits,leds,lineNo,acc) do
    build_output(t,digits,leds,lineNo,acc <> elem(leds[h],lineNo))
  end  
end

File: Erlang/led_numbers.erl
-module(led_Numbers).
-export([showLED/1]).

showLED(Number) when is_integer(Number) ->
	Digits = [ X - 48 || X <- integer_to_list(Number) ],
    Leds = #{0 => {" _  ","| | ","|_| "},
      	     1 => {"  ","| ","| "},
	     2 => {" _  "," _| ","|_  "},
	     3 => {"_  ","_| ","_| "},
             4 => {"    ","|_| ","  | "},
             5 => {" _  ","|_  "," _| "},
	     6 => {" _  ","|_  ","|_| "},
	     7 => {"_   "," |  "," |  "},
	     8 => {" _  ","|_| ","|_| "},
	     9 => {" _  ","|_| "," _| "}},
	build_output(Digits,Digits,Leds,1,"").

build_output([],_,_,3,Acc) -> 
	io:format("~s~n",[Acc]);
build_output([],Digits,Leds,LineNo,Acc) -> 
    build_output(Digits,Digits,Leds,LineNo+1,Acc++"\n");
build_output([H|T],Digits,Leds,LineNo,Acc) -> 
	build_output(T, Digits,Leds,LineNo,Acc++element(LineNo,maps:get(H,Leds))).
